Bio/SeqAnalysis/SeqAnalyzer.py

```diff
@@ -0,0 +1,168 @@
+        flexibility = [sum(__) / len(__) for __ in [_.flexibility() for _ in l_objects]]
+        secondary_structure_fraction = [_.secondary_structure_fraction() for _ in l_objects]
+        seq_gravy = [_.gravy() for _ in l_objects]
+        mol_ext_coefficient = [_.molar_extinction_coefficient()[0] for _ in l_objects]
+        return molecular_weight, pi, amino_acids, aromacity, amino_acids_percent, instability, flexibility, \
+               secondary_structure_fraction, seq_gravy, mol_ext_coefficient
+
+    def n_and_c_term(self) -> tuple:
+        """
+        Counts occurrences of given aa on 'C' and 'N' terminus of protein
+
+        :return: returns a list of two dicts, where list[0] is a dictionary with information about occurrences
+        of aa on 'N' terminus, and list[1] about 'C' terminus respectively
+        """
+        list_c_term = [_[-1] for _ in self.dict.values()]
+        dict_c_term = {i: list_c_term.count(i) / len(set(list_c_term)) for i in set(list_c_term)}
+
+        list_n_term = [_[0] for _ in self.dict.values()]
+        dict_n_term = {i: list_n_term.count(i) / len(set(list_n_term)) for i in set(list_n_term)}
+        return dict_n_term, dict_c_term
+
+    def __init__(self, records):
+        self.dict = records  # Analyzer.makedict(self, records)  # creates a dict with seq ids and seqeances itself
+        self.tuples = [(id, seq) for id, seq in self.dict.items()]
+        self.stdev, self.average, self.median = Analyzer.median_stdev_avg(self)
+        self.ids = Analyzer.id(self)
+        self.lenths = Analyzer.seq_lenth(self)
+        self.objects = Analyzer.molecular_weight(self)
+
+        self.molecular_weight, \
+        self.pi, \
+        self.amino_acids, \
+        self.aromaticity, \
+        self.amino_acids_percent, \
+        self.instability, \
+        self.flexibility, \
+        self.secondary_structure_fraction, \
+        self.gravy, \
+        self.mol_ext_coeff \
+        = Analyzer.molecular_weight(self)
+
+        self.n_term, self.c_term = Analyzer.n_and_c_term(self)
+
+    def results(self):
+        dict_1 = {
+            'median_id': self.median,
+            'mean_id': self.average,
+            'sd_id': self.stdev,
+            'id': self.ids,
+            'sequence_length': self.lenths,
+            'molecular_weight': self.molecular_weight,
+            'theoretical_pl': self.pi,
+            'instability_index': self.instability,
+            'flexibility': self.flexibility,
+            'secondary_structure_fraction': self.secondary_structure_fraction,
+            'extinction_coefficient': self.mol_ext_coeff,
+            'gravy': self.gravy,
+            'c_term_freq': self.c_term,
+            'n_term_freq': self.n_term,
+            'amino_acid_comp': self.amino_acids,
+            'amino_acid_percent': self.amino_acids_percent,
+            'aromacity': self.aromaticity
+        }
+
+        return dict_1
+
+
+Analyzer = Analyzer(records)
```

Comment:
It's very, very bad practice. You should never create object like this. This class usage must be:

```
analyzer = Analyzer(records)
print(analyzer.results())
```

Reply:
Also just to remind, it should be `SeqAnalyzer`, so:

```
analyzer = SeqAnalyzer(records)
print(analyzer.results())
```

Reply:
And at the end, it shouldn't be even here, test data should be in test script